engine/src/main/java/org/terasology/physics/bullet/BulletPhysics.java

```diff
@@ -790,128 +831,151 @@ public boolean needsCollision(BroadphaseProxy proxy0) {
+        public Matrix4f getWorldTransform() {
+            return rb.getWorldTransform();
         }
 
         @Override
-        public org.terasology.math.geom.Vector3f getAngularVelocity(org.terasology.math.geom.Vector3f out) {
-            Vector3f vm = VecMath.to(out);
-            rb.getAngularVelocity(vm);
-            out.set(vm.x, vm.y, vm.z);
-            return out;
+        public Matrix4f setWorldTransform(Matrix4f trans) {
+            rb.setWorldTransform(trans);
+            return trans;
         }
 
         @Override
-        public void setLinearVelocity(org.terasology.math.geom.Vector3f value) {
-            rb.setLinearVelocity(VecMath.to(value));
+        public Vector3f getLinearVelocity(Vector3f out) {
+            return rb.getLinearVelocity();
         }
 
         @Override
-        public void setAngularVelocity(org.terasology.math.geom.Vector3f value) {
-            rb.setAngularVelocity(VecMath.to(value));
+        public Vector3f getAngularVelocity(Vector3f out) {
+            return rb.getAngularVelocity();// out;
         }
 
         @Override
-        public void setOrientation(org.terasology.math.geom.Quat4f orientation) {
-            rb.getWorldTransform(pooledTransform);
-            pooledTransform.setRotation(VecMath.to(orientation));
-            rb.proceedToTransform(pooledTransform);
+        public void setLinearVelocity(Vector3f value) {
+            rb.setLinearVelocity(value);
         }
 
         @Override
-        public void setLocation(org.terasology.math.geom.Vector3f location) {
-            rb.getWorldTransform(pooledTransform);
-            pooledTransform.origin.set(VecMath.to(location));
-            rb.proceedToTransform(pooledTransform);
+        public void setAngularVelocity(Vector3f value) {
+            rb.setAngularVelocity(value);
         }
 
         @Override
-        public void setVelocity(org.terasology.math.geom.Vector3f linear, org.terasology.math.geom.Vector3f angular) {
-            rb.setLinearVelocity(VecMath.to(linear));
-            rb.setAngularVelocity(VecMath.to(angular));
+        public void setOrientation(Quaternionf orientation) {
+            Matrix4f transform = rb.getWorldTransform();
+            rb.setWorldTransform(new Matrix4f().translationRotateScale(transform.getTranslation(new Vector3f()), orientation, 1.0f));
         }
 
         @Override
-        public void setTransform(org.terasology.math.geom.Vector3f location, org.terasology.math.geom.Quat4f orientation) {
-            rb.getWorldTransform(pooledTransform);
-            pooledTransform.origin.set(VecMath.to(location));
-            pooledTransform.setRotation(VecMath.to(orientation));
-            rb.proceedToTransform(pooledTransform);
+        public void setLocation(Vector3f location) {
+            Matrix4f translation = rb.getWorldTransform();
+            Quaternionf quaternion = new Quaternionf().setFromUnnormalized(translation);
+
+//            Quat4f quaternion = new Quat4f();
```

Comment:
More commented out code - either remove if it is no longer needed or put a comment / TODO to explain why it is still there please :-)

Seeing at least one more of these, will skip commenting on any more